src/dynarec/arm64/dynarec_arm64_0f.c

```diff
@@ -439,7 +462,75 @@ uintptr_t dynarec64_0F(dynarec_arm_t* dyn, uintptr_t addr, uintptr_t ip, int nin
                     SADDLPQ_16(v1, v0);
                     SQXTN_16(q0, v1);
                     break;
-
+                case 0x05:
+                    INST_NAME("PHSUBW Gm,Em");
+                    nextop = F8;
+                    GETGM(q0);
+                    GETEM(q1, 0);
+                    v0 = fpu_get_scratch(dyn);
+                    VUZP1_16(v0, q0, q1);
+                    VUZP2_16(q0, q0, q1);
+                    VSUB_16(q0, v0, q0);
+                    break;
+                case 0x06:
+                    INST_NAME("PHSUBD Gm,Em");
+                    nextop = F8;
+                    GETGM(q0);
+                    GETEM(q1, 0);
+                    v0 = fpu_get_scratch(dyn);
+                    VUZP1_32(v0, q0, q1);
+                    VUZP2_32(q0, q0, q1);
+                    VSUB_32(q0, v0, q0);
+                    break;
+                case 0x07:
+                    INST_NAME("PHSUBSW Gm,Em");
+                    nextop = F8;
+                    GETGM(q0);
+                    GETEM(q1, 0);
+                    v0 = fpu_get_scratch(dyn);
+                    VUZP1_16(v0, q0, q1);
+                    VUZP2_16(q0, q0, q1);
+                    SQSUB_16(q0, v0, q0);
+                    break;
+                case 0x08:
+                    INST_NAME("PSIGNB Gm,Em");
+                    nextop = F8;
+                    GETGM(q0);
+                    GETEM(q1, 0);
+                    v0 = fpu_get_scratch(dyn);
+                    v1 = fpu_get_scratch(dyn);
+                    CMGT_0_8(v0, q1);
+                    VAND(v0, v0, q0);
+                    CMLT_0_8(v1, q1);
+                    VMUL_8(q0, q0, v1);
+                    VORR(q0, q0, v0);
+                    break;
+                case 0x09:
+                    INST_NAME("PSIGNW Gm,Em");
+                    nextop = F8;
+                    GETGM(q0);
+                    GETEM(q1, 0);
+                    v0 = fpu_get_scratch(dyn);
+                    v1 = fpu_get_scratch(dyn);
+                    CMGT_0_16(v0, q1);
+                    VAND(v0, v0, q0);
+                    CMLT_0_16(v1, q1);
+                    VMUL_16(q0, q0, v1);
+                    VORR(q0, q0, v0);
+                    break;
+                case 0x0A:
+                    INST_NAME("PSIGND Gm,Em");
+                    nextop = F8;
+                    GETGM(q0);
+                    GETEM(q1, 0);
+                    v0 = fpu_get_scratch(dyn);
+                    v1 = fpu_get_scratch(dyn);
+                    CMGT_0_32(v0, q1);
+                    VAND(v0, v0, q0);
+                    CMLT_0_32(v1, q1);
+                    VMUL_32(q0, q0, v1);

```

Comment:
and same here, VAND instead of VUML_32

Reply:
oh, 111111 is taken as -1! Ah yeah, I remember now. Yeah, ok, looks good indeed!